File: Components/Strata.Client.Core/main/java/strata/client/core/presenter/IModelStore.java
//////////////////////////////////////////////////////////////////////////////
// IModelStore.java
//////////////////////////////////////////////////////////////////////////////

package strata.client.core.presenter;

public
interface IModelStore
{
    <M> IModelStore
    attach(IUpdatable<M> updatable);

    <M> IModelStore
    detach(IUpdatable<M> updatable);

    <M> IModelStore
    insert(Class<M> key,M model);

    <M> IModelStore
    remove(Class<M> key);

    <M> boolean
    hasUpdatable(Class<M> key);

    <M> boolean
    hasModel(Class<M> key);

    <M> void
    apply(IAction<M> action);
}

//////////////////////////////////////////////////////////////////////////////

File: Components/Strata.Client.Core/main/typescript/Presenter/IModelStore.ts
import {IUpdatable} from "./IUpdatable";
import {IAction} from "./IAction";

export
interface IModelStore
{
    attach<M>(updatable: IUpdatable<M>): IModelStore;

    detach<M>(updatable: IUpdatable<M>): IModelStore;

    insert<M>(key: string,model:M): IModelStore;

    remove(key: string): IModelStore;

    hasUpdatable(key: string): boolean;

    hasModel(key: string): boolean;

    apply<M>(action: IAction<M>): void;

}